src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2707,27 +2732,13 @@ private void EmitConstantExpression(TypeSymbol type, ConstantValue constantValue
         private void EmitInitObj(TypeSymbol type, bool used, SyntaxNode syntaxNode)
         {
             if (used)
-            {
-                if (type.IsPointerType() || type.SpecialType == SpecialType.System_UIntPtr)
```

Comment:
The purpose is that EmitInitObj does just that - emits "initobj". 
The optimizations will be done in the InitDefaultValue. We should do all the optimizations in one place.

If callers can benefit from the optimization they should call that. 

I will follow up on all the callers in a separate change. We were likely missing some optimizations already since some of them were already done only in InitDefaultValue. Did not want to mix unrelated change into this since it could affect unrelated testcases.

---
In reply to: [118782391](https://github.com/dotnet/roslyn/pull/19713#discussion_r118782391) [](ancestors = 118782391)